### ISSUE TITLE
"Directory already exists" when trying to clone git repo for first time

### ISSUE THREAD

**Issue by quvide** [zeapo/Android-Password-Store#122]
When trying to clone a repo for the first time, it gives a message "Directory already exists". When I select "delete directory" it gives "Message from jgit: Attempt to get length of null array".

**Comment by mpnordland** [comment]
I have a similar problem, i choose the recommended hidden option, and then enter my ssh server details and then attempt to clone. It prompts to delete the directory, which I agree to, but then nothing happens.

**Comment by pdroalves** [comment]
Same problem here.

**Comment by pdroalves** [comment]
I solved my problem. For some stupid reason I was trying to use my public ssh key instead of the private key. I replaced it and it worked.

As suggestion, we need an alert about using an invalid ssh key.

**Comment by powellc** [comment]
I just double checked all my information and the key I was using and nothing is working. Just sits there after asking to delete the current directory :(

**Comment by zeapo** [comment]
Thank you all for the report. We will change the behavior.

On the side note, we do not remove everything, the text is wrong. We remove the content of the store then the user asks us to in **settings > delete store**

**Comment by zeapo** [comment]
I deleted the previous comment as it was intended for another issue. Sorry.

Are you trying to clone into the internal store?

**Comment by mpnordland** [comment]
Yes. I worked around this issue by downgrading and messing around with the older version. I originally experienced this again, but later it worked.
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by tuxick** [comment]
Same problem here, first "Directory already exists", telling it to delete and then the Internal Exception: "Message from jgit: Attempt to get length of null array"

**Comment by bjmgeek** [comment]
I had the same issue.  I have multiple user profiles on my android device (LG Nexus 5 with Android 6.0.1) and I thought it may have been related to that (I already have password store installed on my main user profile, and I was trying to install on a new user).

**Comment by adskankster** [comment]
I'm getting the same issue - have done for a few versions, now. It did work a while back, but I had to re-create my repo on my server and when I tried to re-create it on my phone, I got this error. It doesn't matter what dirs or repo locations I choose, I always get the directory already exists issue - but I haven't seen the internal exception mentioned above, fwiw.

Running on aliendalvik on a Jolla, but as I say, it did work originally.

**Comment by adskankster** [comment]
I'm getting the same issue - have done for a few versions, now. It did work a while back, but I had to re-create my repo on my server and when I tried to re-create it on my phone, I got this error. It doesn't matter what dirs or repo locations I choose, I always get the directory already exists issue - but I haven't seen the internal exception mentioned above, fwiw.

Running on aliendalvik on a Jolla, but as I say, it did work originally.

**Comment by adskankster** [comment]
I'm getting the same issue - have done for a few versions, now. It did work a while back, but I had to re-create my repo on my server and when I tried to re-create it on my phone, I got this error. It doesn't matter what dirs or repo locations I choose, I always get the directory already exists issue - but I haven't seen the internal exception mentioned above, fwiw.

Running on aliendalvik on a Jolla, but as I say, it did work originally.
